platform/linux-generic/pktio/netmap.c

```diff
@@ -786,6 +824,100 @@ static int netmap_recv(pktio_entry_t *pktio_entry, int index,
 	return num_rx;
 }
 
+static int netmap_recv_tmo(pktio_entry_t *pktio_entry, int index,
+			   odp_packet_t pkt_table[], int num, uint64_t usecs)
+{
+	struct timeval timeout;
+	int ret;
+	int maxfd;
+	fd_set readfds;
+
+	ret = netmap_recv(pktio_entry, index, pkt_table, num);
+	if (ret != 0)
+		return ret;
+
+	/* 32 bit architectures use 32-bit division if possible */
+	if (sizeof(size_t) < 8) {
+		if (usecs <= UINT32_MAX) {
+			timeout.tv_sec = ((uint32_t)usecs) / (1000 * 1000);
```

Comment:
To me the value of this optimization seems questionable. System calls are used here anyway, so one division operation (next  comment) shouldn't have that much effect.

Reply:
Have you tried 64-bit division on 32-bit ARM? It's really slow! I was surprised how slow it is until I tested it on my Raspberry Pi. ARM does not have a 32-bit division... But perhaps your next comment will make the optimization a bit less necessary and it thus could be removed.

Reply:
Some measurements: empty select() with zero timeout operates at 1.2 M ops / second, 64-bit division operates at 8.8 M ops / second on my Raspberry Pi 3 Model B with the optimization in your next comment. The problem is that in the current code the division is done quite early, but I think I could restructure it so that the division is done only whenever absolutely needed for select(). So yes, I think you may be right.

Reply:
Looks good now. With more modern/bigger CPUs the division should be faster. Just for curiosity, If you share your benchmark code I can run it on A72.

Reply:
Sure, https://github.com/jmtilli/divbenchmark

It gives me this on my Raspberry Pi 3 Model B running on 32-bit mode:

    $ make unit
    div
    123 456789
    1.14user 0.00system 0:01.15elapsed 99%CPU (0avgtext+0avgdata 1044maxresident)k
    0inputs+0outputs (0major+73minor)pagefaults 0swaps
    divfast
    123 456789
    0.13user 0.00system 0:00.13elapsed 96%CPU (0avgtext+0avgdata 1092maxresident)k
    0inputs+0outputs (0major+74minor)pagefaults 0swaps
    select
    2.40user 6.60system 0:09.00elapsed 99%CPU (0avgtext+0avgdata 752maxresident)k
    0inputs+0outputs (0major+69minor)pagefaults 0swaps